### ISSUE TITLE
markup5ever aborts in build.rs

### ISSUE THREAD

**Issue by emi2k01** [bjorn3/rustc_codegen_cranelift#1209]
Repro:

```
git clone https://github.com/servo/html5ever --depth 1 && \
    cd html5ever/markup5ever && \
    cargo_cf build
```
It seems to crash in `write_to(&mut generated)` (grep it) when string cache [string-cache](https://github.com/servo/string-cache) calls `AtomType::to_tokens` in the body of `write_to`.

I don't know how to get the backtrace since it just aborts.

It works correctly with normal cargo.

**Comment by bjorn3** [comment]
Thanks for the issue. Markup5ever uses proc_macro2 in it's build script, which tries to detect if it s running inside a proc macro or not by invoking a function that panics when not run inside a proc macro. cg_clif doesn't support catching panics and thus aborts. This should have been fixed in https://github.com/dtolnay/proc-macro2/pull/300, but it seems that I made a mistake with the `#[cfg]`. Opened https://github.com/dtolnay/proc-macro2/pull/303 with a confirmed fix.

Duplicate of https://github.com/bjorn3/rustc_codegen_cranelift/issues/1101

**Comment by bjorn3** [comment]
This should be fixed now as of proc-macro 1.0.33.
